blog/_posts/2017-09-04-gsoc-NeuralNetDiffEq.md

```diff
@@ -0,0 +1,172 @@
+
+Also after implementing the solver for ODEs (Ordinary Differential Equations) and System of ODEs the difficult part was to make the Neural Network converge, especially for the system of ODEs. As there are a lot of factors involved in Neural Networks like Hidden Layer Width, Number of Hidden Neurons, Activations, Weights etc., I relied on my Machine Learning background as well as the help from my mentors to experiment with most of the feasible settings of NN hyper-parameters and recording the accuracy of convergence and performance of the solver.
+
+Making NN converge for System of ODEs is not as easy as it seems and took up most of the time for experimentation and tuning. Predicting the system of DEs solution with larger domain is still a challenge which needs to be worked upon.
+
+## Implementation and Work
+Implementation involved integration of Mathematical and Machine Learning aspects through programming and to build a solver for ODEs. The [DiffEqBase library](https://github.com/JuliaDiffEq/DiffEqBase.jl) is used as a base to extend the Algorithm and solver interface. The work done till now can be seen on the [github repository](https://github.com/JuliaDiffEq/NeuralNetDiffEq.jl) primarily in [this branch](https://github.com/JuliaDiffEq/NeuralNetDiffEq.jl/tree/SingleNN_Approach). This work involves implementing a Neural Network solver for ODEs with customized iterpolation based on NN prediction. The solver also works for system of ODEs but needs to be optimized for larger domains.
+
+### How does it work?
+We need to construct a trial solution for our Differential Equation in terms of the NN output which should also satisfy the DE boundary conditions.  
+We define a loss function for the Neural Network which is same as equating the Differential to zero and minimizing that function (closer to 0 better convergence) wuth the trial solution substituted in it in place of the original function (or the solution to the DE). The Neural Network uses Adam optimization algorithm for backpropagation to tune its weights. GPU operation is supported by using the KnetArray type (the arry type used in [KNet.jl](https://github.com/denizyuret/Knet.jl)) instead of regular Array for parameters and data, but requires CUDNN driver installed to access GPU hardware.
```

Comment:
To do so, we define ...

Reply:
equating the derivative to zero

Reply:
The opening sentence in general is a little confusing. I would be more direct: the loss function is the difference between the derivative of the neural net solution and the true derivative defined by the ODE.

Reply:
"GPU operation is supported" odd construction.